File: leetcode/15.3/1.cpp
#include<bits/stdc++.h>
using namespace std;
bool check(string s) {
        string rev = s;
        reverse(rev.begin(), rev.end());
        return s == rev;
    }
    string longestPalindrome(string s) {
        int l = 0, r = 0;
        string ans, cur;
        cur.resize(2);
        for (int i = 0; i < s.size(); i++) {
            r = s.size() - i - 1;
            cur += s[l] + s[r];
            cout << cur << endl;
            if (check(cur)) {
                l = i;
                r = r;
                break;
            }
        }
		cout << l << " " << r;
        return ans;
    }
int main() {
	string s;
	cin >> s;
	cout << longestPalindrome(s) << endl;
}

File: leetcode/15.3/1.py
class Solution:
	def check(self, s: str) -> bool:
		return s[::-1] == s

	def longestPalindrome(self, s: str) -> str:
		r_cur: int = 0
		i_cur: int = 0

		if (len(s) < 2):
			return s
		
		if (len(s) == 2):
			if (s[0] == s[1]):
				return s
			return s[0]

		for i in range(len(s)):
			for r in range(len(s) - 1, 0, -1):
				cur: str = s[i] + s[r]
				if (self.check(cur)):
					if (abs(i - r) > abs(i_cur - r_cur)):
						r_cur = r
						i_cur = i
		return s[i_cur:r_cur + 1]


s: str = input()
print(Solution().longestPalindrome(s))